db/deals.go

```diff
@@ -272,9 +291,51 @@ func (d *DealsDB) List(ctx context.Context, query string, cursor *graphql.ID, of
 		whereArgs = append(whereArgs, searchArgs...)
 	}
 
+	if filter != nil {
+		if where != "" {
```

Comment:
I think we need to check if filterWhere == "" also

Reply:
I can add it to be safe but I don't think it should be possible to get to that state, as if filter is not nil it should always yield a valid query statement.